lib/io.toit

```diff
@@ -544,55 +514,66 @@ class BufferedReader implements Reader:
       reader.peek_string 5  // >> Error!
   ```
   */
-  peek-string n -> string:
+  peek-string n/int -> string:
     // Fast case.
     if n == 0: return ""
-    if buffered >= n:
-      first := arrays_.first
+    if buffered-size >= n:
+      first := buffered_.first
       end := first-array-position_ + n
       if first.size >= end:
         return first.to-string first-array-position_ end
     // Slow case.
-    return (bytes n).to-string
-
-  /** Deprecated. */
-  read-word -> string:
-    return read-until ' '
+    return (peek-bytes n).to-string
 
   /**
   Reads a line as a string.
 
-  Lines are terminated by a newline character (`'\n'`) except for the final
-    line.
-  Carriage returns (`'\r'`) are removed from lines terminated by `'\r\n'`.
+  If $keep-newline is true, the returned string includes the newline character.
+  If $keep-newline is false, trims the trailing '\r\n' or '\n'. This method
+    removes a '\r' even if the platform is not Windows. If the '\r' needs to be
+    preserved, set $keep-newline to true and remove the trailing '\n' manually.
```

Comment:
The next PR has a bit more documentation what happens if the last line doesn't end with a new line.